src/main/kotlin/com/terraformation/backend/nursery/db/BatchStore.kt

```diff
@@ -124,7 +126,10 @@ class BatchStore(
     val subLocationIds = fetchSubLocationIds(batchId)
     val totalWithdrawn = getTotalWithdrawn(batchId)
 
-    return ExistingBatchModel(batchesRow, subLocationIds, totalWithdrawn)
+    val accessionId = batchesRow.accessionId
+    val accessionNumber = if (accessionId != null) fetchAccessionNumber(accessionId) else null
```

Comment:
This is totally fine as is, but FYI, another way to write it would be
```kotlin
val accessionNumber = batchesRow.accessionId?.let { fetchAccessionNumber(it) }
```

Reply:
Ah yes, much better